src/lib.rs

```diff
@@ -260,39 +274,53 @@ pub enum EncodingKind {
 #[derive(Debug)]
 pub struct Extract<'a> {
     source: &'a path::Path,
-    archive: ArchiveKind,
-    encoding: EncodingKind,
+    archive: Option<ArchiveKind>,
 }
 impl<'a> Extract<'a> {
     pub fn from_source(source: &'a path::Path) -> Extract<'a> {
         Self {
             source: source,
-            archive: ArchiveKind::Plain,
-            encoding: EncodingKind::Plain,
+            archive: None,
         }
     }
     pub fn archive(&mut self, kind: ArchiveKind) -> &mut Self {
-        self.archive = kind;
+        self.archive = Some(kind);
         self
     }
-    pub fn encoding(&mut self, kind: EncodingKind) -> &mut Self {
-        self.encoding = kind;
-        self
-    }
-    pub fn extract_into(&self, into_dir: &path::Path) -> Result<()> {
+    pub fn extract_into(&self, into_dir: &path::Path, file_to_extract: &path::Path) -> Result<()> {
```

Comment:
Could you implement this as a new function (`extract_file` maybe?) so we can keep the semantics of `extract_into` extracting the whole archive?